sfepy/discrete/functions.py

```diff
@@ -42,6 +43,54 @@ def set_function(self, function, is_constant=False):
     def set_extra_args(self, **extra_args):
         self.extra_args = extra_args
 
+def make_sfepy_function(fun_or_name=None):
+    """
+    Convenience decorator to quickly create
+    :class:`sfepy.discrete.functions.Function` objects.
+
+    Has two modes of use either without parameter::
+
+        @make_sfepy_function
+        def my_function(...):
+            ...
+
+    or with name::
+
+        @make_sfepy_function("new_name_for_my_function")
+        def my_function(...):
+            ...
+
+    Parameters
+    ----------
+    fun_or_name : string, optional
+        Name to be saved within `Function` instance,
+        if None name of decorated function is used.
+
+    Returns
+    -------
+    `sfepy.discrete.functions.Function`
+        With attribute name set to provided name or original function name.
+    """
+    if callable(fun_or_name):
+        return Function(fun_or_name.__name__, fun_or_name)
```

Comment:
Why is this branch actually needed? Is it not covered by `return Function(fun.__name__, fun)` in `functionizer()`?

Reply:
This allows usage where you omit not only parameters but also parenthesis and use decorator like in the first mode of use in documentation commentary:
```python
@make_sfepy_function
def my_function(...):
            ...
```

Reply:
I see, thanks!